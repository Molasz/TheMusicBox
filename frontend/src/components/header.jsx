import React from 'react';
import './header.scss';

import { Link } from 'react-router-dom';

function header({ isLogged }) {
  return (
    <header className='header'>
      <Link className='header__logo' to='/' />

      <input className='header__input'></input>

      {isLogged ? (
        <img
          src='https://upload.wikimedia.org/wikipedia/commons/thumb/b/b2/Hamburger_icon.svg/1200px-Hamburger_icon.svg.png'
          alt='manu icon'
          className='header__menu'
        />
      ) : (
        <Link to='/login' className='header__login'>
          <img
            src={require('../assets/login.png')}
            alt='login icon'
            className='login__icon'
          />
        </Link>
      )}
    </header>
  );
}

export default header;
