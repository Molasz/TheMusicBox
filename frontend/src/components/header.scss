.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;

  background-color: black;

  height: 80px;

  padding: 0 20px;
  .header__logo {
    margin: 0 -25px;
    width: 25%;
  }
  .header__input {
    flex: 0.5;
    height: 35px;
    padding: 0 30px;

    font-size: 20px;

    background-image: url(https://image.flaticon.com/icons/svg/25/25313.svg);
    background-size: 20px;
    background-position: 5px;
    background-repeat: no-repeat;
  }
  .header__menu {
    width: 40px;
  }
}
