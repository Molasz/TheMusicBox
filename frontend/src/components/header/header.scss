.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;

  background-color: black;

  height: 80px;

  padding: 0 20px;

  min-width: 500px;
  .header__logo {
    width: 140px;
    height: 100%;

    background-image: url(../../assets/logo_white_little.png);
    background-size: 100%;
    background-repeat: no-repeat;
    background-position: center;
  }
  .header__input {
    width: 40%;
    height: 35px;
    padding: 0 30px;

    font-size: 20px;

    background-image: url(https://image.flaticon.com/icons/svg/25/25313.svg);
    background-size: 20px;
    background-position: 5px;
    background-repeat: no-repeat;
  }
  .header__login {
    display: flex;
    align-items: center;
    justify-content: center;
    .login__icon {
      width: 40px;
    }
  }
  .header__menu {
    position: relative;
    display: inline-block;
    .menu__icon {
      color: white;
      font-size: 40px;
    }
    .menu__content {
      display: none;
      position: absolute;
      background-color: #000000;
      padding: 10px 3px;
      z-index: 1;
      .content__list {
        display: flex;
        flex-direction: column;
        justify-content: center;
        justify-items: center;

        .list__item {
          margin-bottom: 10px;
        }
      }
    }
  }
  .header__menu:hover .menu__content {
    display: block;
  }
}

@media (min-width: 768px) {
  .header {
    .header__logo {
      width: 300px;
      background-image: url(../../assets/logo_white.png);
    }
  }
}
