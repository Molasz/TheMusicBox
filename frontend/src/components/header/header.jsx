import React, { useState, useEffect } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import './header.scss';

import store from '../../redux/store';
import { searchBand } from '../../redux/actions/bandActions';

import LoginButton from './loginButton';
import LogoutButton from './logoutButton';

import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';

function Header({ auth, band }) {
  const [redirect, setRedirect] = useState(null);

  useEffect(() => {
    setRedirect(null);
  });

  function onSearch(event) {
    if (event.keyCode === 13) {
      store.dispatch(searchBand(event.target.value.toLowerCase()));
      event.target.value = '';
      setRedirect(<Redirect to='/search' />);
    }
  }

  return (
    <header className='header'>
      {redirect}
      <Link className='header__logo' to='/' />

      <div className='header__input'>
        <input className='input' onKeyUp={onSearch}></input>
        <SearchIcon className='icon__search' />
        <ArrowBackIosIcon className='icon__arrow' />
      </div>

      {auth ? (
        <div className='header__menu'>
          <MenuIcon className='menu__icon' />
          <div className='menu__content'>
            <div className='content__list'>
              <Link to='/profile' className='list__item'>
                Profile
              </Link>
              <LogoutButton />
            </div>
          </div>
        </div>
      ) : (
        <LoginButton className='login__icon' />
      )}
    </header>
  );
}

export function mapStateToProps(state) {
  return {
    auth: state.authReducer.userIdentifier,
    band: state.bandReducer.search
  };
}

export default connect(mapStateToProps, null)(Header);
