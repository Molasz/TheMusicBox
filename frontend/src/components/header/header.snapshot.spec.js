import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Header from './header';

import { mapStateToProps } from './header';

jest.mock('mapStateToProps');

// Problemes amb l'snapshot a compos que utilitzen redux
describe('Header snapshot', () => {
  it('Should match with userIdentifier', () => {
    mapStateToProps.mockReturnValue(1);
    const tree = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should match without userIdentifier', () => {
    mapStateToProps.mockReturnValue(null);
    const tree = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
