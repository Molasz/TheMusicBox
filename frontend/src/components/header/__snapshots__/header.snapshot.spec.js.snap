// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot Should match with userIdentifier 1`] = `null`;

exports[`Header snapshot Should match without userIdentifier 1`] = `null`;
